docs/guides/vm-architectures/README.md

```diff
@@ -0,0 +1,399 @@
+
+**Pros:**
+
+* **Guaranteed data consistency.** This approach ensures that any successful query returns 100% of the requested data. If vmselect receives only a partial response from its vmstorage nodes, the entire query is marked as failed, preventing any misleading or incomplete results.
+
+**Cons:**
+
+* **Lower Availability.** This approach sacrifices availability to guarantee consistency. So if more than replicationFactor vmstorage nodes are unavailable, read queries will start returning errors.
+
+### Buffering Strategy Trade-off
+
+Once you have a vmagent sending data to the storage component (vmsingle or cluster), you face your first important trade-off: what should vmagent do when the storage is temporarily unavailable? This choice defines the trade-off between higher availability (by not losing data) and lower resource consumption (by not using disk). By default, vmagent acts as a durable queue: it persists compressed unsent data to the local filesystem. The size of the queue is controlled via \`--remoteWrite.maxDiskUsagePerURL\` and can be [estimated in advance](https://docs.victoriametrics.com/victoriametrics/vmagent/#calculating-disk-space-for-persistence-queue).
+
+**Path A: Stateful Mode (Most Reliable).**  By default, [the operator uses ephemeral storage](https://docs.victoriametrics.com/operator/resources/vmagent/#statefulmode) for the vmagent queue. In production, we recommend explicitly configuring a PersistentVolumeClaim (PVC) for vmagent to ensure the buffer is stored on a persistent disk and survives pod restarts. [The documentation](https://docs.victoriametrics.com/victoriametrics/vmagent/#on-disk-persistence) about on-disk persistence.
+
+**Pros:**
+
+* **Improved reliability.** Unsent data is safe during vmagent restarts or when remote storage is down (until queue is full).
+
+**Cons:**	
+
+* **Requires additional resources.** Deployment becomes stateful, uses disk space, and I/O. The queue size can build extra pressure on remote storage once it becomes available.
+
+For Enterprise users, the queueing can be offloaded to an external message broker, such as **Kafka**. In that case vmagent can [read or write into Kafka](https://docs.victoriametrics.com/victoriametrics/integrations/kafka/).
+
+**Path B: Ephemeral Buffering (with tmpfs).** For maximum performance, the vmagent buffer directory can be mounted as a tmpfs volume, which is physically stored in the node's RAM. In Kubernetes, this is configured via `emptyDir: { medium: "Memory" }`.
+
+**Pros:**
+
+* **Fast I/O.** Buffering happens at RAM speed. This path safeguards against brief network outages without any loss of performance.
+
+**Cons:**
+
+* **Significant risk of data loss.** Unsent data is lost on vmagent restarts. The queue size is limited by the available memory.
```

Comment:
```suggestion
* **increased risk of data loss.** Unsent data is lost on vmagent restarts. The queue size is limited by the available memory.
```
I don't think we should categorize risk, but we should mention the increase in probability